### PR TITLE
Feature: Add update script to add unique IDs derived from external source

### PR DESCRIPTION
**Pull request by joel-wright**

### CODE REVIEW

File: backend/scripts/add_unique_ids/add_unique_ids.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from app.db.models import Document
+from app.db.session import SessionLocal
+
+
+LookupID = Mapping[str, str]
+
+
+def load_id_map() -> Mapping[str, str]:
+    current_folder = pathlib.Path(__file__).parent.absolute()
+    id_map_path = current_folder / "ids_map.json"
```
Comment (diversemix) [inline backend/scripts/add_unique_ids/add_unique_ids.py:17]: Where does this file come from? Is it worth adding a README on how it should be used?

File: backend/scripts/add_unique_ids/extra/map_ids.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            if (name, url) in current_name_urls:
+                print(name)
+            current_name_urls[(name, url)] = r["id"]
+
+    doc_id_to_import = {}
+    with open(cclw_newer_path) as cclw_newer:
+        csv_reader = DictReader(cclw_newer)
+        for r in csv_reader:
+            name = r["Title"].strip()
+            url = r["Documents"].split("|")[0].strip().replace("http://", "https://")
```
Comment (diversemix) [inline backend/scripts/add_unique_ids/extra/map_ids.py:23]: why do we do this?

File: backend/scripts/add_unique_ids/add_unique_ids.py (hunk 1 above)
Comment (joel-wright) [reply backend/scripts/add_unique_ids/add_unique_ids.py:17]: I really consider this script a one-shot; everything after here should have the IDs and there's no guarantee that what I did this time would continue to work...

File: backend/scripts/add_unique_ids/extra/map_ids.py (hunk 2 above)
Comment (joel-wright) [reply backend/scripts/add_unique_ids/extra/map_ids.py:23]: because the CCLW input file is pure bonkers 🤣

**Review by diversemix — APPROVED** [review]
lgtm

**Pull request merged**